src/AsyncTCP.cpp

```diff
@@ -226,15 +259,23 @@ static void _async_service_task(void* pvParameters) {
   for (;;) {
     if (_get_async_event(&packet)) {
 #if CONFIG_ASYNC_TCP_USE_WDT
-      if (esp_task_wdt_add(NULL) != ESP_OK) {
-        log_e("Failed to add async task to WDT");
+      switch((size_t)packet->arg){
+        case LWIP_TCP_RECV:
+        case LWIP_TCP_POLL:
+          if (esp_task_wdt_add(NULL) != ESP_OK) {
+            log_e("Failed to add async task to WDT");
+          }
+          _handle_async_event(packet);
+          if (esp_task_wdt_delete(NULL) != ESP_OK) {
```

Comment:
There are things weird related to the TWDT:
- It is only activated if `CONFIG_ASYNC_TCP_RUNNING_CORE` is not set. If `CONFIG_ASYNC_TCP_RUNNING_CORE=1` is set (i.e. what I use to do), then there is no watchdog.
- If `CONFIG_ASYNC_TCP_USE_WDT` is set, `CONFIG_ASYNC_TCP_USE_WDT=1` then can to be set if a watchdog is active on ore 1

All that to say that by default, TWDT is activated and otherwise it is user-controled.

So should the calls to `esp_task_wdt_add` and `esp_task_wdt_delete` be only around the use callbacks instead of here wrapping also the AsyncTCP code ?

I don't know how costly it is to call `esp_task_wdt_add` and `esp_task_wdt_delete` frequently, but it seems quite inefficient... These things should only be there to protect against long user callbacks right ?

Reply:
`LWIP_TCP_POLL ` => goes to onPolll callback, which is controlled by ESpAsyncWS internally and dos not leak to the user

`LWIP_TCP_RECV ` => onData => this is the one that could trigger some user code callback.

So IMO, we could just surround this part of code with `esp_task_wdt_add` and `esp_task_wdt_delete ` ?

```
      if (_recv_cb) {
        _recv_cb(_recv_cb_arg, this, b->payload, b->len);
      }
```

Reply:
yeah, I do not like this enable/disable either. Better I will rework it in a bit different way. And also clean up this define mess